DM: add v1.0.2 release notes (#1716)
pingcap/docs: dev/reference/tools/data-migration/releases/1.0.2.md

---
title: DM 1.0.2 Release Notes
category: Releases
---

# DM 1.0.2 Release Notes

Release date: October 30, 2019

DM version: 1.0.2

DM-Ansible version: 1.0.2

## Improvements

- Generate some config items for DM-worker automatically
- Generate some config items for replication task automatically

> - Simplify the output of query-status without arguments

Reviewer: ```suggestion
- Simplify the output of `query-status` without arguments
```

- Manage DB connections directly for downstream

## Bug fixes

- Fix some panic when starting up or executing SQL statements
- Fix abnormal sharding DDL replication on DDL execution timeout
- Fix starting task failure caused by the checking timeout or any inaccessible DM-worker
- Fix SQL execution retry for some error

> ## Detailed Bug Fixes and Changes

Reviewer 2: ```suggestion
## Detailed bug fixes and changes
```

> - Generate random server-id for DM-worker config automatically [#337](https://github.com/pingcap/dm/pull/337)

Reviewer: ```suggestion
- Generate random `server-id` for DM-worker config automatically [#337](https://github.com/pingcap/dm/pull/337)
```

How about quoting (``` ` ```) the words that have been quoted in https://github.com/pingcap/docs-cn/pull/1995/files.

Author: Thanks for reminding me. Updated now.

- Generate `flavor` for DM-worker config automatically [#328](https://github.com/pingcap/dm/pull/328)
- Generate `relay-binlog-name` and `relay-binlog-gtid` for DM-worker config automatically [#318](https://github.com/pingcap/dm/pull/318)

> - Generate table name list for dumping in task config from black & white table lists automatically [#326](https://github.com/pingcap/dm/pull/326)

Reviewer 2: ```suggestion
- Generate the name list of tables to be dumped in task config from black & white table lists automatically [#326](https://github.com/pingcap/dm/pull/326)
```

- Fix potential DM-worker panic when restoring subtask from local meta [#311](https://github.com/pingcap/dm/pull/311)
- Fix DM-worker panic when committing a DML transaction failed [#313](https://github.com/pingcap/dm/pull/313)
- Fix DM-worker or DM-master panic when the listening port is being used [#301](https://github.com/pingcap/dm/pull/301)
- Fix retry for error code 1105 [#321](https://github.com/pingcap/dm/pull/321), [#332](https://github.com/pingcap/dm/pull/332)
- Fix retry for `Duplicate entry` and `Data too long for column` [#313](https://github.com/pingcap/dm/pull/313)
- Fix task check timeout when having large amounts of tables in upstream [#327](https://github.com/pingcap/dm/pull/327)
- Fix starting task failure when any DM-worker is not accessible [#319](https://github.com/pingcap/dm/pull/319)
- Fix potential DM-worker startup failure in GTID mode after being recovered from corrupt relay log [#339](https://github.com/pingcap/dm/pull/339)
- Fix in-memory TPS count for sync unit [#294](https://github.com/pingcap/dm/pull/294)
- Manage DB connections directly for downstream [#325](https://github.com/pingcap/dm/pull/325)

> - Improve error system by refining error information passed between components [#320](https://github.com/pingcap/dm/pull/320)

Reviewer 2: ```suggestion
- Improve the error system by refining error information passed between components [#320](https://github.com/pingcap/dm/pull/320)
```
